### ISSUE TITLE
json-schema converter

### ISSUE THREAD

**Issue by amcdnl** [PLEEROCK/validator.ts#5]
It would be nice to have a json-schema converter like this: https://github.com/lightsofapollo/joi-to-json-schema

**Comment by PLEEROCK** [comment]
yes I was thinking about this as an additional feature for this library. 

But I think most of people will use decorator based validation, because having schemas in separate files makes duplication (property names are duplicated both in the model and schema) + there is no refactoring benefits, and if your model's property name changed - you must manually change property name in the schema too.

But I think someone will still may need this feature, one of the use case that really works if user have an object without class, but still wants to perform schema-based validation.

I'll add accepting PRs label for now, until I return back to this library

**Comment by amcdnl** [comment]
I want it for swagger docs and json data generators that use json schema.

**Comment by TomMarius** [comment]
A great feature would be to be able to create a JSON-Schema object from a decorated class.

**Comment by pleerock** [comment]
feel free to contribute 😃

**Comment by xe4me** [comment]
@TomMarius @pleerock has anyone made any progress on this ?

**Comment by epiphone** [comment]
I started a little library for JSON Schema conversion [here](https://github.com/epiphone/class-validator-jsonschema). Any feedback would be appreciated!

**Comment by lamuertepeluda** [comment]
Hi there, I was wondering if it would be possible to accept [standard JSON Schema](https://json-schema.org/) as input (much like AJV) in addition to the proposed schema definitions of this library?

**Comment by jgburet** [comment]
https://www.npmjs.com/package/class-validator-jsonschema

**Comment by braaar** [comment]
Hi, @amcdnl, we have now started using the Discussions tab, which should be used to discuss and mature ideas. This thread has seen very little activity recently, so it would be preferable if you could close this issue and open a new thread in the discussion section if there is still some interest in the idea.

**Comment by braaar** [comment]
I'll convert this to a discussion

**Comment by NoNameProvided** [comment]
I consider this to be out of scope for the project. (Also schema validation is deprecated in the library.) However, this should like a good idea for a user-contributed library.

Closing this as wontfix. 

If the issue still persists, you may open a new Q&A in the [discussions tab](https://github.com/typestack/class-validator/discussions/categories/questions-answers) and someone from the community may be able to help.
